### PR TITLE
Set llvm configs when building lld

### PR DESCRIPTION
**Pull request by topjohnwu**
Several LLVM configs applied when building LLVM should also be applied when building LLD.

For example, without the fix, setting both `llvm.thin-lto=true` and `rust.lld=true` when building on Linux will fail, since we need to manually override the compiler and linker to `clang`, but those will not be used when building LLD, causing link errors.

r? @jyn514

### CODE REVIEW

File: src/bootstrap/native.rs [hunk 1]
```diff
@@ -789,6 +789,17 @@ impl Step for Lld {
             .define("LLVM_CONFIG_PATH", llvm_config_shim)
             .define("LLVM_INCLUDE_TESTS", "OFF");
 
+        if builder.config.llvm_thin_lto {
+            cfg.define("LLVM_ENABLE_LTO", "Thin");
+            if !target.contains("apple") {
+                cfg.define("LLVM_ENABLE_LLD", "ON");
```
Comment (jyn514) [inline src/bootstrap/native.rs:795]: Why is apple special-cased here? why does LTO imply LLD?

File: src/bootstrap/native.rs [hunk 2]
```diff
@@ -789,6 +789,17 @@ impl Step for Lld {
...
+            }
+        }
+
+        if let Some(ref linker) = builder.config.llvm_use_linker {
+            cfg.define("LLVM_USE_LINKER", linker);
+        }
```
Comment (jyn514) [inline src/bootstrap/native.rs:801]: This is duplicated from the `Llvm` step. Can we add a function that applies the LLVM config to both Llvm and Lld so we ensure they stay in sync?

**Comment by jyn514** [comment]
@topjohnwu let me know when you have time to look at this again - I left a couple comments.

**Comment by topjohnwu** [comment]
@jyn514 thanks for the reminder, I'll check this out soon

File: src/bootstrap/native.rs (hunk 1 above)
Comment (topjohnwu) [reply src/bootstrap/native.rs:795]: In Rust's build setup, it uses ThinLTO (`-flto=Thin`), this is a LLVM specific feature so require the usage of LLD.

Apple's Xcode does not ship with LLD, instead linking is done using its own linker `ld64`, which supports thin LTO out of the box (it kinda make sense as Apple's toolchain is based on LLVM).

File: src/bootstrap/native.rs (hunk 2 above)
Comment (topjohnwu) [reply src/bootstrap/native.rs]: Move the code into its own helper function as requested

File: src/bootstrap/native.rs (hunk 1 above)
Comment (jyn514) [reply src/bootstrap/native.rs]: Ok. Can you add that as a comment please?

Comment (topjohnwu) [reply src/bootstrap/native.rs]: Added comments as requested.

**Comment by jyn514** [comment]
Thanks!

@bors r+ rollup

**Pull request merged**